### PR TITLE
CLN remove 'import pandas as pd' from pandas/core/generic.py

### PR DESCRIPTION
**Pull request by MarcoGorelli**
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them

---

will do the rest (excluding `pandas/tests`) and add a check if this is the kind of thing we want

### CODE REVIEW

File: pandas/core/generic.py [hunk 1]
```diff
@@ -5304,7 +5304,11 @@ def sample(
                         "when sampling from a Series."
                     )
 
-            weights = pd.Series(weights, dtype="float64")
+            if isinstance(self, ABCSeries):
+                func = self._constructor
+            else:
+                func = self._constructor_sliced
```
Comment (MarcoGorelli) [inline pandas/core/generic.py:5310]: Or is it better to import `Series` within this function so as to avoid an if-else block?

**Comment by jbrockmendel** [comment]
nice

File: pandas/core/generic.py (hunk 1 above)
Comment (jreback) [reply pandas/core/generic.py:5310]: yeah you can just import Series here

Comment (jreback) [reply pandas/core/generic.py:5310]: the only reason you woul want this is to handle a subclass

Comment (jreback) [reply pandas/core/generic.py:5310]: but i guess this is ok

**Comment by jreback** [comment]
thanks @MarcoGorelli

**Pull request merged**